### PR TITLE
Handle maxLength for userEvent.type

### PR DESCRIPTION
**Pull request by klujanrosas**
This is related to #161 .
I took the liberty to implement it. I look forward to your comments or suggestions.

### CODE REVIEW

**Comment by klujanrosas** [comment]
@afontcu I also can't think of a reason why they would fail, but one can never be too cautious!

I will find some time to add those tests as well, thank you

**Comment by klujanrosas** [comment]
@afontcu sorry to bother you, could this be reviewed again?

**Comment by kentcdodds** [comment]
Are we sure the browser doesn't continue to fire input events even though the maxlength is reached? We don't have to actually update the value, but we should make sure we fire all the same events the browser does.

**Comment by calebeby** [comment]
@kentcdodds I just tested an input with a maxlength, it looks like it only fires `input` events till the max length is reached.

**Comment by kentcdodds** [comment]
Cool. Are we still firing keyup/keydown events?

**Comment by calebeby** [comment]
I'm pretty sure that we should be, but I'm not sure whether it does in this PR

**Comment by klujanrosas** [comment]
@kentcdodds @calebeby  that makes sense, I will update this PR so it still fires keyup/keydown events, thank you for your comments

**Comment by carlos-cne** [comment]
Hi! I arrived here because I have an input with a max-length attribute.
When I call userEvent.type(...) with a string more than max-length, it does not work and I got a string with more than max-length.

I'm testing if input.value is equal to max-length
`expect(inputValue.value).toHaveLength(60);`

**Comment by kentcdodds** [comment]
Hi @carlos-cne,

Yes, once this PR is updated and then merged that issue should go away.

**Comment by klujanrosas** [comment]
@Gpx @kentcdodds I've updated this to reflect what was suggested in the comments above, let me know if there's anything else that I should cover or change :hatching_chick:

File: __tests__/vue/type.js [hunk 1]
```diff
@@ -145,4 +145,48 @@ describe("userEvent.type", () => {
...
+
+      userEvent.type(inputEl, text, {
+        allAtOnce: true
+      });
+
+      expect(inputEl).toHaveProperty("value", slicedText);
+      expect(input).toHaveBeenCalledTimes(1);
+      expect(keydown).not.toHaveBeenCalled();
+      expect(keypress).not.toHaveBeenCalled();
+      expect(keyup).not.toHaveBeenCalled();
```
Comment (kentcdodds) [inline __tests__/vue/type.js]: Wait, shouldn't these all be called several times?

Comment (klujanrosas) [reply __tests__/vue/type.js]: not for `allAtOnce` afaik

Comment (kentcdodds) [reply __tests__/vue/type.js]: Oh

Oh! lol, my bad.

**Comment by kentcdodds** [comment]
@all-contributors please add @klujanrosas for code and tests

**Pull request merged**